packages/synapse-sdk/src/telemetry/service.ts

```diff
@@ -0,0 +1,218 @@
+   * This is optional and can be set by the user via the synapse.telemetry.sentry.setContext() method.
+   * If not set, the SDK will use 'synapse-sdk' as the default app name.
+   */
+  appName?: string
+  tags?: Record<string, string> // optional: custom tags
+}
+
+/**
+ * Configuration for runtime detection and context
+ */
+interface RuntimeContext {
+  sdkVersion: string
+  runtime: 'browser' | 'node'
+  network: 'mainnet' | 'calibration'
+  userAgent?: string
+  appName?: string
+}
+
+export interface DebugDump {
+  lastEventId: string | undefined
+  events: any[]
+}
+
+const { Sentry, integrations } = await getSentry()
+
+/**
+ * Main telemetry service that manages the adapter and provides high-level APIs
+ */
+export class TelemetryService {
+  // private adapter: TelemetryAdapter
+  private enabled: boolean
+  private context: RuntimeContext
+  private eventBuffer: any[] = []
+  private readonly maxBufferSize = 50
+
+  readonly sentry: SentryType
+
+  constructor(config: TelemetryConfig, context: RuntimeContext) {
+    const resolvedConfig = resolveTelemetryConfig(config)
+    this.enabled = resolvedConfig.enabled
+    this.context = context
+
+    this.sentry = Sentry
+    if (this.enabled) {
+      this.initSentry()
+    }
+  }
+  private initSentry(): void {
+    this.sentry.init({
+      dsn: 'https://3ed2ca5ff7067e58362dca65bcabd69c@o4510235322023936.ingest.us.sentry.io/4510235328184320',
+      // Setting this option to false will prevent the SDK from sending default PII data to Sentry.
+      // For example, automatic IP address collection on events
+      sendDefaultPii: false,
+      release: `@filoz/synapse-sdk@v${this.context.sdkVersion}`,
+      beforeSend: this.sanitizeError.bind(this),
+      // Enable tracing/performance monitoring
+      tracesSampleRate: 1.0, // Capture 100% of transactions for development (adjust in production)
+      // Integrations configured per-runtime in sentry-dep files
+      integrations,
+    })
+
+    // things that we don't need to search for in sentry UI, but may be useful for debugging should be set as context
+    this.sentry.setContext('environment', {
+      userAgent: this.context.userAgent, // not useful for searching, but may be useful for debugging
+    })
+
+    // things that we can search in the sentry UI (i.e. not millions of unique potential values, like userAgent would have) should be set as tags
+    this.sentry.setTag('appName', this.context.appName ?? 'synapse-sdk') // should only be a few values
+    this.sentry.setTag('synapseSdkVersion', `@filoz/synapse-sdk@v${this.context.sdkVersion}`) // useful to know which version of the SDK is being used
+    this.sentry.setTag('runtime', this.context.runtime) // only two values, useful for searching
+    this.sentry.setTag('network', this.context.network as 'mainnet' | 'calibration') // only two values, useful for searching
+  }
+
+  protected sanitizeError(event: any): any {
+    this.addToBuffer(event)
+
+    return event
+  }
```

Comment:
I think we can remove this and just have the caller call "addToBuffer" directly.